client/src/main.c

```diff
@@ -1208,15 +1228,29 @@ int eventFilter(void * userdata, SDL_Event * event)
       switch(event->window.event)
       {
         case SDL_WINDOWEVENT_ENTER:
+          if (g_state.wminfo.subsystem == SDL_SYSWM_WAYLAND)
```

Comment:
Why is this not just part of the != X11 codepath?

Reply:
This is Wayland-specific logic to set `g_cursor.inView`. Normally this is done by `handleMouseNormal`. For the Wayland-specific logic, this needs to be handled by `SDL_WINDOWEVENT_ENTER` and `SDL_WINDOWEVENT_EXIT`.